### PR TITLE
Add nym function

### PR DESCRIPTION
**Pull request by lhh**

### CODE REVIEW

File: tests/test_strutil.py [hunk 1]
```diff
@@ -23,3 +23,15 @@ def test_normal(self):
         self.assertEqual(list_or_splitstr('''a
     b        c'''), ['a', 'b', 'c'])
         self.assertEqual(list_or_splitstr('a	b	c'), ['a', 'b', 'c'])
+
+
+class test_nym(unittest.TestCase):
```
Comment (fuzzball81) [inline tests/test_strutil.py:28]: suggestion non-blocking: Might be good to add a test that covers the '\t' case.

**Review by fuzzball81 — APPROVED** [review]
Just a few non-blocking suggestions, otherwise looks good.

File: toolchest/strutil.py [hunk 2]
```diff
@@ -30,3 +30,9 @@ def regex_match(pattern, arg):
     if re.match(pattern, arg) is not None:
         return True
     return False
+
+
+def nym(arg):
```
[the remark this answers is not in the document]
Comment (jguiditta) [reply toolchest/strutil.py:35]: suggestion: Agree, please add docstring here. Also, does 'nym' mean something specific? I quick search gets me "New York Mets baseball", which I doubt its the intent. Maybe a name the just describes what it does would be more obvious (and if I am missing the obvious, feel free to explain in the doc string).

File: tests/test_strutil.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import unittest
 
-from toolchest.strutil import list_or_splitstr
+from toolchest.strutil import list_or_splitstr, nym
 
 
 class test_list_or_splitstr(unittest.TestCase):
```
Comment (jguiditta) [inline tests/test_strutil.py:6]: suggestion: I realize this library has a number of older tests that still use unittest, but for new code, I would rather see us use pytest, like the rest of our projects (and newer code here).

**Review by jguiditta — COMMENTED** [review]
Overall, looks reasonable, just a few requests for tweaks.

File: tests/test_strutil.py (hunk 3 above)
Comment (lhh) [reply tests/test_strutil.py]: Done

File: tests/test_strutil.py (hunk 1 above)
Comment (lhh) [reply tests/test_strutil.py]: Done

File: toolchest/strutil.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lhh) [reply toolchest/strutil.py:35]: Done

**Review by fuzzball81 — DISMISSED** [review]

File: toolchest/strutil.py [hunk 4]
```diff
@@ -30,3 +30,24 @@ def regex_match(pattern, arg):
...
+    low. For example, if we want to search a dictionary, we'd check the
+    nym of the value provided with the nym of the key to see if they
+    match. This should not be used when likelihood of collisions is high.
+    (Origin: Greek word meaning "name")
+
+    Parameters:
+        arg (string): A string to create the nym for
+
+    Returns:
+        ret (string): A lower-case string with whitespace swapped to _
```
Comment (jguiditta) [inline toolchest/strutil.py:49]: flake8 error here:

  flake8: commands[0]> flake8 setup.py docs toolchest tests
  toolchest/strutil.py:49:71: W291 trailing whitespace

**Review by jguiditta — COMMENTED** [review]
Other than the minor flake8 error, looks good to me! Thanks for making those tweaks, they help a lot.

**Review by jguiditta — APPROVED** [review]

**Review by fuzzball81 — APPROVED** [review]

**Pull request merged**